Application/src/view/visuals/component/ClassRelationship.java

```diff
@@ -107,45 +140,50 @@ else if (XDistance < 0 && YDistance > 0) {
             offsetIncrementY *= (-1);
         }
 
-        // To solve the issue with all the different cases of linking the classes, we used Bresenham’s Line Algorithm
-        // http://csunplugged.org/wp-content/uploads/2014/12/Lines.pdf
-        int A, B, P;
-
         // This is used to find out whether to use the Y axis or the X axis as a parameter for the for loop
-        // this avoids having 2 loops
         int stepsParameter;
-
+        int A, B, P; // Variables used for Bresenham’s Line Algorithm
         int gap = 0; // Used to remove steps from the for loops so that the arrow appears correctly
 
         /// Using the Bresenham’s Line Algorithm on X axis as reference
         if (Math.abs(XDistance) > Math.abs(YDistance)) {
-            stepsParameter = Math.abs(XDistance);
+            int[] params = initBresenhamsParameters(XDistance, YDistance);
+            stepsParameter = params[0];
+            A = params[1];
+            B = params[2];
+            P = params[3];
 
-            A = 2 * Math.abs(YDistance);
-            B = A - 2 * Math.abs((XDistance));
-            P = A - Math.abs((XDistance));
-            // decreasing the gap to show arrow correctly when it is a horizontal line
-            if (YDistance == 0) {
-                gap -= 2;
-            }
             // decreasing the gap to show arrow correctly when it is a horizontal line pointing right
-            if (XDistance != 0) {
-                gap -= 1;
+            if (XDistance > 0 && YDistance == 0) {
+                gap -= 3;
+            }
+            // decreasing the gap to show arrow correctly when it is a horizontal line pointing left
+            if (XDistance < 0 && YDistance == 0) {
+                gap -= 2;
             }
         }
-        // Using the Bresenham’s Line Algorithm on the Y axis as reference
+        // Using the Bresenham’s Line Algorithm on the Y axis as reference by swapping the parameters
         else {
-            stepsParameter = Math.abs(YDistance);
+            int[] params = initBresenhamsParameters(YDistance, XDistance);
+            stepsParameter = params[0];
+            A = params[1];
+            B = params[2];
+            P = params[3];
 
-            A = 2 * Math.abs(XDistance);
-            B = A - 2 * Math.abs((YDistance));
-            P = A - Math.abs((YDistance));
-            // decreasing the gap to show arrow correctly when it is a vertical line line
-            if (XDistance == 0) {
+            // decreasing the gap to show arrow correctly when it is a vertical line pointing down
+            if (XDistance == 0 && YDistance > 0) {
+                gap -= 2;
+            }
+            // decreasing the gap to show arrow correctly when it is a vertical line pointing up
+            if (XDistance == 0 && YDistance < 0) {
```

Comment:
Sorry but can you please explain why you are using different logic for the gap in the if and else

Reply:
Im talking about this entire function

Reply:
@timjon No pattern here, i used TDD to optimize each gap

Reply:
What would the difference be if the logic for the gaps is the same for both cases?
And how would it look without the gap variable?

Reply:
@timjon as you can see there are 6 cases, 2 for each line (horizontal, vertical diagonal) , without the gap variable the inheritance arrow would either look far away from the island, or hide behind it.

Reply:
Can you make some pictures of the 2 different cases i asked about and pictures on how it looks with your current code?
I know its annoying but I need to be sure that it accually has to be like this since its a pain to read because of all the cases, and it would be much simpler for everyone to read and understand if there wasnt that many cases

Reply:
http://oi65.tinypic.com/29kqs5z.jpg
so this is testing diagonal lines, the gap variable here is used in 2 cases (relationship from top to bot and relationship from bot to top). these 2 cases are different so each of them decrease the gap variable by certain number.
without the gap variable at all, the arrow would be buried underneath the island because relationship is pointing at the coordinates of the island itself, not the edge of the island

Reply:
Is the picture you linked drawn using the current variables?

Reply:
yes with different placements of the classes @timjon